src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,120 @@
+  [master-request-handler ca-request-handler]
+  (compojure/routes
+    (compojure/GET "/node/*" request (master-request-handler request))
+    (compojure/GET "/facts/*" request (master-request-handler request))
+    (compojure/GET "/file_content/*" request (master-request-handler request))
+    (compojure/GET "/file_metadatas/*" request (master-request-handler request))
+    (compojure/GET "/file_metadata/*" request (master-request-handler request))
+    (compojure/GET "/file_bucket_file/*" request (master-request-handler request))
+    ;; Coercing Content-Type to "application/octet-stream" for Puppet 4
+    ;; compatibility.
+    (compojure/PUT "/file_bucket_file/*" request
+      (master-request-handler
+        (-> request
+          (assoc :content-type "application/octet-stream") ; deprecated in ring >= 1.2
+          (assoc-in [:headers "content-type"] "application/octet-stream")))) ; ring >= 1.2 spec
+    (compojure/HEAD "/file_bucket_file/*" request (master-request-handler request))
+    (compojure/GET "/catalog/*" request (master-request-handler request))
+    (compojure/POST "/catalog/*" request (master-request-handler request))
+    (compojure/PUT "/report/*" request (master-request-handler request))
+    (compojure/GET "/resource_type/*" request (master-request-handler request))
+    (compojure/GET "/resource_types/*" request (master-request-handler request))
+    (compojure/GET "/status/*" request (master-request-handler request))
+
+    ;; TODO: when we get rid of the legacy dashboard after 3.4, we should remove
+    ;; this endpoint as well.  It makes more sense for this type of query to be
+    ;; directed to PuppetDB.
+    (compojure/GET "/facts_search/*" request (master-request-handler request))
+
+    ;; legacy CA routes
+    (compojure/ANY "/certificate_status/*" request (ca-request-handler request))
+    (compojure/ANY "/certificate_statuses/*" request (ca-request-handler request))
+    (compojure/GET "/certificate/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_revocation_list/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_request/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_requests/*" request (ca-request-handler request))))
+
+(defn legacy-routes
+  "Creates all of the compojure routes for the master."
+  [master-request-handler ca-request-handler]
+  (compojure/routes
+    (compojure/context "/v2.0" request
+                       (v2_0-routes master-request-handler))
+    (compojure/context "/:environment" [environment]
+                       (legacy-master-routes master-request-handler ca-request-handler))
+    (route/not-found "Not Found")))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Public
+
+(defn build-ring-handler
+  [master-handler master-mount master-api-version
+   ca-handler ca-mount ca-api-version]
+  (log/info "Building ring handler for legacy service")
```

Comment:
Might be best to put this line above where the ring handler is registered for the first time and not here.  Otherwise, looks like we get this in the output every time a request is processed.